### PR TITLE
Collapsible code examples in documentation

### PR DESCRIPTION
**Pull request by jeffin143**
Handling : https://github.com/mlpack/ensmallen/issues/139

:rocket:  Done

### CODE REVIEW

File: doc/callbacks.md [hunk 1]
```diff
@@ -8,53 +8,74 @@ as:
 
 Callbacks can be passed as an argument to the `Optimize()` function:
 
+<details>
+<summary>Click to check out the example</summary>
```
Comment (rcurtin) [inline doc/callbacks.md]: The `kramdown` Markdown processor, which is what's used for ensmallen.org, unfortunately doesn't work with this... for some reason the closing tag has to be on a separate line:

```
<summary>Click to check out the example
</summary>
```

That's kind of irritating, but, well, it's what we have to do I guess. :)

File: doc/callbacks.md [hunk 2]
```diff
@@ -8,53 +8,74 @@ as:
...
+<p>
```
Comment (rcurtin) [inline doc/callbacks.md]: The `<p>` isn't necessary, I have the styling working with out.  So we can remove these. :)

File: doc/callbacks.md [hunk 3]
```diff
@@ -8,53 +8,74 @@ as:
...
+
 ```c++
-RosenbrockFunction f;
-arma::mat coordinates = f.GetInitialPoint();
+  RosenbrockFunction f;
+  arma::mat coordinates = f.GetInitialPoint();
```
Comment (rcurtin) [inline doc/callbacks.md]: I don't think the spacing changes are needed. :+1:

File: doc/callbacks.md [hunk 4]
```diff
@@ -8,53 +8,74 @@ as:
...
+  optimizer.Optimize(f, coordinates, PrintLoss());
 ```
 
+</p>
+</details>  
+
 Passing multiple callbacks is just the same as passing a single callback:
 
+<details>
+<summary>Click to check out the example</summary>
```
Comment (rcurtin) [inline doc/callbacks.md]: Another thought, do you think we can change this text to `Click to collapse/expand example code.`?  That way the message makes sense even when it's expanded.

Comment (zoq) [reply doc/callbacks.md]: Agreed, ``Click to collapse/expand example code.`` is more descriptive.

Comment (conradsnicta) [reply doc/callbacks.md]: I also vote for `Click to collapse/expand example code`

Comment (jeffin143) [reply doc/callbacks.md]: Done :) :rocket:

File: doc/callbacks.md (hunk 3 above)
Comment (jeffin143) [reply doc/callbacks.md]: :) adapted

File: doc/callbacks.md (hunk 2 above)
Comment (jeffin143) [reply doc/callbacks.md]: Sure

File: doc/callbacks.md (hunk 1 above)
Comment (jeffin143) [reply doc/callbacks.md]: :smiley:

File: doc/optimizers.md [hunk 5]
```diff
@@ -1231,15 +1357,21 @@ Note that the `NesterovMomentumUpdate` class has the constructor
...
+<summary>Click to collapse/expand example code.
+</summary>
+
 ```c++
 RosenbrockFunction f;
 arma::mat coordinates = f.GetInitialPoint();
 
 NesterovMomentumSGD optimizer(0.01, 32, 100000, 1e-5, true,
-    MomentumUpdate(0.5));
+  MomentumUpdate(0.5));
```
Comment (rcurtin) [inline doc/optimizers.md]:
```suggestion
    MomentumUpdate(0.5));
```

Oops, I think there was no need to change this one. :)

Comment (jeffin143) [reply doc/optimizers.md]: should I change this ?

**Comment by jeffin143** [comment]
> Hello there! Thanks for your contribution. I see that this is your first contribution to mlpack. If you'd like to add your name to the list of contributors in `src/mlpack/core.hpp` and `COPYRIGHT.txt` and you haven't already, please feel free to push a change to this PR---or, if it gets merged before you can, feel free to open another PR.
> 
> In addition, if you'd like some stickers to put on your laptop, I'd be happy to help get them in the mail for you. Just send an email with your physical mailing address to [stickers@mlpack.org](mailto:stickers@mlpack.org), and then one of the mlpack maintainers will put some stickers in an envelope for you. It may take a few weeks to get them, depending on your location. +1

I know @mlpack-bot :), I have been contributing often :-p

**Comment by rcurtin** [comment]
Yeah, @mlpack-bot doesn't look across repositories... I was too lazy to get that support working right. :)  Sorry about that...

**Comment by rcurtin** [comment]
Awesome, thanks! :+1:

**Pull request merged**